plugins/flytekit-omegaconf/README.md

```diff
@@ -0,0 +1,79 @@
+# Flytekit OmegaConf Plugin
+
+Flytekit python natively supports serialization of many data types for exchanging information between tasks.
+The Flytekit OmegaConf Plugin extends these by the `DictConfig` type from the
+[OmegaConf package](https://omegaconf.readthedocs.io/) as well as related types
+that are being used by the [hydra package](https://hydra.cc/) for configuration management.
+
+## Task example
+```
+from dataclasses import dataclass
+import flytekitplugins.omegaconf  # noqa F401
+from flytekit import task, workflow
+from omegaconf import DictConfig
+
+@dataclass
+class MySimpleConf:
+    _target_: str = "lightning_module.MyEncoderModule"
+    learning_rate: float = 0.0001
+
+@task
+def my_task(cfg: DictConfig) -> None:
+    print(f"Doing things with {cfg.learning_rate=}")
+
+
+@workflow
+def pipeline(cfg: DictConfig) -> None:
+    my_task(cfg=cfg)
+
+
+if __name__ == "__main__":
+    from omegaconf import OmegaConf
+
+    cfg = OmegaConf.structured(MySimpleConf)
+    pipeline(cfg=cfg)
+```
+
+## Transformer configuration
+
+The transformer can be set to one of three modes:
+
+`Dataclass` - This mode should be used with a StructuredConfig and will reconstruct the config from the matching dataclass
+during deserialisation in order to make typing information from the dataclass and continued validation thereof available.
+This requires the dataclass definition to be available via python import in the Flyte execution environment in which
+objects are (de-)serialised.
+
+`DictConfig` - This mode will deserialize the config into a DictConfig object. In particular, dataclasses are translated
+into DictConfig objects and only primitive types are being checked. The definition of underlying dataclasses for
+structured configs is only required during the initial serialization for this mode.
+
+`Auto` - This mode will try to deserialize according to the Dataclass mode and fall back to the DictConfig mode if the
+dataclass definition is not available. This is the default mode.
+
+To set the mode either initialise the transformer with the `mode` argument or set the mode of the config directly:
+
+```python
+from flytekitplugins.omegaconf.config import SharedConfig, OmegaConfTransformerMode
+from flytekitplugins.omegaconf import DictConfigTransformer
+
+# Set the mode directly on the transformer
```

Comment:
I find it untypical that users import type transformers to configure their behaviour. Would the user have to register `transformer_slim` in this example with the `TypeEngine` again?

Would it be ok for you to remove/not document this option but use the auto mode by default and only document how it can be overridden by configuring the dict config itself?

Reply:
Or is `SharedConfig` a singleton that is used to control the behaviour globally?

Reply:
> Do I understand it correctly that these are two mechanisms that achieve exactly the same thing?

I think the second example configures the mode of the (dict) transformer globally, whereas the first one only configures the specific instance.

Either way, I can't think of a use-case where this would really be useful, I think we can configure it globally across all transformers, the way you suggested. Any objections @SebS94 ?
Will come back to you once I implement it (ran out of time now).

Reply:
Addressed in https://github.com/flyteorg/flytekit/pull/2299/commits/e6e18964ff9d0f53a6722d1320e239f3d3a96545

We provide global mode and a context manager for ability to control it locally still.